common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package com.gradle;
+
+import com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension;
+import com.gradle.enterprise.gradleplugin.GradleEnterprisePlugin;
+import com.gradle.scan.plugin.BuildScanExtension;
+import com.gradle.scan.plugin.BuildScanPlugin;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.initialization.Settings;
+
+public class CommonCustomUserDataGradlePlugin implements Plugin<Object> {
+    public void apply(Object target) {
+        if (target instanceof Settings) {
+            applySettingsPlugin((Settings) target);
+        } else if (target instanceof Project) {
+            applyProjectPlugin((Project) target);
+        }
+    }
+
+    private void applyProjectPlugin(Project project) {
+        project.getPlugins().withType(BuildScanPlugin.class, __ -> {
+            BuildScanExtension buildScan = project.getExtensions().getByType(GradleEnterpriseExtension.class).getBuildScan();
+            enhanceBuildScan(buildScan, project.getRootProject());
+        });
+    }
+
+    private void applySettingsPlugin(Settings settings) {
+        settings.getPlugins().withType(GradleEnterprisePlugin.class, __ ->
+                settings.getGradle().projectsLoaded(gradle -> {
+                    BuildScanExtension buildScan = settings.getExtensions().getByType(GradleEnterpriseExtension.class).getBuildScan();
+                    enhanceBuildScan(buildScan, gradle.getRootProject());
+                })
+        );
+    }
+
+    private void enhanceBuildScan(BuildScanExtension buildScanExtension, Project rootProject) {
+        enhanceBuildScan(new BuildScanEnhancer(buildScanExtension, rootProject));
+    }
+
+    private void enhanceBuildScan(BuildScanEnhancer buildScanEnhancer) {
+        buildScanEnhancer.tagOs();
+        buildScanEnhancer.tagIde();
+        buildScanEnhancer.tagCiOrLocal();
+        buildScanEnhancer.addCiMetadata();
+        buildScanEnhancer.addGitMetadata();
+        buildScanEnhancer.addTestParallelization();
+        buildScanEnhancer.addTestSystemProperties();
+    }
```

Comment:
Maybe a matter of taste, but I would probably move these invocations to the enhancer to keep it closer to what we do for the Maven extension.